spec/ddtrace/contrib/delayed_job/app.rb

```diff
@@ -0,0 +1,34 @@
+require 'active_record'
+require 'delayed_job'
+require 'delayed_job_active_record'
+
+logger = Logger.new(STDOUT)
+logger.level = Logger::INFO
+
+ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
```

Comment:
I know we've done this a lot in previous test files to configure ActiveRecord with Minitest, but we really shouldn't be establishing database connections simply by virtue of calling `require 'app'`. I would consider wrapping this into a module, function, or RSpec reference that can be optionally composed into specific examples, where necessary.

A big downside is that this currently creates a dependence of some kind of preconfigured database state: if this state changes, it could break tests in an indeterminate number of places across the test suite.

In an ideal world, specs should encapsulate and manage the state they need to run successfully, rather than depending on such globals. If specs need to share some kind of common test setup (as opposed to state: they should never share state), then it makes sense to have this setup in composable module within a separate file like this. It gives the specs greater control over what state gets set, and how/when to set it. And if this state changes, it's easier to track down which examples specifically depend upon it.

Reply:
Also, 💯 for use of in-memory SQLite, rather than depending on an external database service.

Reply:
Agreed `requrie 'app'` is not the best approach. 

I've switched to using RSpec `around` to setup and close the DB for each test.